### PR TITLE
fix(server): Download artifacts from UI. Fixes #4338

### LINKED ISSUE #4338: open artifact : permission denied
## Summary 

**What happened:**
I installed the quick-start-postgres.yaml, submitted the  example using the argo UI. After the example succeeded,I want to see the logs in Archived Workflows, but it shows "Internal Server Error". Then  I clicked on" logs from the artifacts", but it shows "open artifact788239015: permission denied". I clicked on "OUTPUT ARTIFACTS" and I can see the file"main-logs", but when I clicked on it, stil, "open artifact083636081: permission denied"

**What version of Argo Workflows are you running?** 
v2.11.6

argo-server logs
```
time="2020-10-21T13:09:37Z" level=info msg="Download artifact" artifactName=main-logs nodeId=delightful-poochenheimer uid=1b4ba4cd-a7a0-4db8-a5f6-b0b71d4b7fc4
time="2020-10-21T13:13:31Z" level=info msg="Alloc=9032 TotalAlloc=45921 Sys=72384 NumGC=23 Goroutines=79"
```
workflow-controller logs
```
time="2020-10-21T12:58:29Z" level=info msg="Alloc=6195 TotalAlloc=63291 Sys=71872 NumGC=63 Goroutines=181"
time="2020-10-21T12:58:39Z" level=info msg="Performing periodic workflow GC"
time="2020-10-21T12:58:39Z" level=info msg="Zero old offloads, nothing to do"
time="2020-10-21T12:59:50Z" level=info msg="Processing workflow" namespace=argo workflow=delightful-poochenheimer
time="2020-10-21T12:59:50Z" level=info msg="Updated phase  -> Running" namespace=argo workflow=delightful-poochenheimer
time="2020-10-21T12:59:50Z" level=info msg="Pod node delightful-poochenheimer initialized Pending" namespace=argo workflow=delightful-poochenheimer
time="2020-10-21T12:59:50Z" level=info msg="Created pod: delightful-poochenheimer (delightful-poochenheimer)" namespace=argo workflow=delightful-poochenheimer
time="2020-10-21T12:59:50Z" level=info msg="Workflow update successful" namespace=argo phase=Running resourceVersion=161374914 workflow=delightful-poochenheimer
time="2020-10-21T12:59:50Z" level=info msg="Processing workflow" namespace=argo workflow=delightful-poochenheimer
time="2020-10-21T12:59:50Z" level=info msg="Skipped pod delightful-poochenheimer (delightful-poochenheimer) creation: already exists" namespace=argo podPhase=Pending workflow=delightful-poochenheimer
time="2020-10-21T12:59:50Z" level=info msg="Workflow update successful" namespace=argo phase=Running resourceVersion=161374917 workflow=delightful-poochenheimer
time="2020-10-21T12:59:51Z" level=info msg="Processing workflow" namespace=argo workflow=delightful-poochenheimer
time="2020-10-21T12:59:51Z" level=info msg="Updating node delightful-poochenheimer message: ContainerCreating"
time="2020-10-21T12:59:51Z" level=info msg="Skipped pod delightful-poochenheimer (delightful-poochenheimer) creation: already exists" namespace=argo podPhase=Pending workflow=delightful-poochenheimer
time="2020-10-21T12:59:51Z" level=info msg="Workflow update successful" namespace=argo phase=Running resourceVersion=161374944 workflow=delightful-poochenheimer
time="2020-10-21T12:59:52Z" level=info msg="Processing workflow" namespace=argo workflow=delightful-poochenheimer
time="2020-10-21T12:59:52Z" level=info msg="Skipped pod delightful-poochenheimer (delightful-poochenheimer) creation: already exists" namespace=argo podPhase=Pending workflow=delightful-poochenheimer
time="2020-10-21T13:00:07Z" level=info msg="Processing workflow" namespace=argo workflow=delightful-poochenheimer
time="2020-10-21T13:00:07Z" level=info msg="Setting node delightful-poochenheimer outputs"
time="2020-10-21T13:00:07Z" level=info msg="Updating node delightful-poochenheimer status Pending -> Running"
time="2020-10-21T13:00:07Z" level=info msg="Skipped pod delightful-poochenheimer (delightful-poochenheimer) creation: already exists" namespace=argo podPhase=Running workflow=delightful-poochenheimer
time="2020-10-21T13:00:07Z" level=info msg="Workflow update successful" namespace=argo phase=Running resourceVersion=161375296 workflow=delightful-poochenheimer
time="2020-10-21T13:00:07Z" level=info msg="Processing workflow" namespace=argo workflow=delightful-poochenheimer
time="2020-10-21T13:00:07Z" level=info msg="Skipped pod delightful-poochenheimer (delightful-poochenheimer) creation: already exists" namespace=argo podPhase=Running workflow=delightful-poochenheimer
time="2020-10-21T13:00:08Z" level=info msg="Processing workflow" namespace=argo workflow=delightful-poochenheimer
time="2020-10-21T13:00:08Z" level=info msg="Updating node delightful-poochenheimer status Running -> Succeeded"
time="2020-10-21T13:00:08Z" level=info msg="Updated phase Running -> Succeeded" namespace=argo workflow=delightful-poochenheimer
time="2020-10-21T13:00:08Z" level=info msg="Marking workflow completed" namespace=argo workflow=delightful-poochenheimer
time="2020-10-21T13:00:08Z" level=info msg="Marking workflow as pending archiving" namespace=argo workflow=delightful-poochenheimer
time="2020-10-21T13:00:08Z" level=info msg="Checking daemoned children of " namespace=argo workflow=delightful-poochenheimer
time="2020-10-21T13:00:08Z" level=info msg="Workflow update successful" namespace=argo phase=Succeeded resourceVersion=161375322 workflow=delightful-poochenheimer
time="2020-10-21T13:00:08Z" level=info msg="archiving workflow" namespace=argo uid=1b4ba4cd-a7a0-4db8-a5f6-b0b71d4b7fc4 workflow=delightful-poochenheimer
time="2020-10-21T13:00:08Z" level=info msg="Delete pod for gc successfully" namespace=argo pod=delightful-poochenheimer
time="2020-10-21T13:00:08Z" level=info msg="archiving workflow" namespace=argo uid=1b4ba4cd-a7a0-4db8-a5f6-b0b71d4b7fc4 workflow=delightful-poochenheimer
time="2020-10-21T13:00:08Z" level=info msg="Found Workflow argo/delightful-poochenheimer set expire at 2020-10-21 13:05:08 +0000 UTC (4m59.240014881s from now)"
time="2020-10-21T13:00:08Z" level=info msg="Queueing workflow argo/delightful-poochenheimer for delete in 4m59.240014881s"
time="2020-10-21T13:00:08Z" level=info msg="Found Workflow argo/delightful-poochenheimer set expire at 2020-10-21 13:05:08 +0000 UTC (4m59.239699386s from now)"
time="2020-10-21T13:00:08Z" level=info msg="Queueing workflow argo/delightful-poochenheimer for delete in 4m59.239699386s"
time="2020-10-21T13:03:29Z" level=info msg="Alloc=6258 TotalAlloc=66707 Sys=72128 NumGC=66 Goroutines=181"
time="2020-10-21T13:03:39Z" level=info msg="Performing periodic workflow GC"
time="2020-10-21T13:03:39Z" level=info msg="Zero old offloads, nothing to do"
time="2020-10-21T13:05:08Z" level=info msg="Deleting TTL expired workflow argo/delightful-poochenheimer"
time="2020-10-21T13:05:08Z" level=info msg="Successfully deleted 'argo/delightful-poochenheimer'"
time="2020-10-21T13:05:08Z" level=info msg="Found Workflow argo/delightful-poochenheimer set expire at 2020-10-21 13:05:08 +0000 UTC (-36.460904ms from now)"
time="2020-10-21T13:05:08Z" level=info msg="Queueing workflow argo/delightful-poochenheimer for delete in 0s"
time="2020-10-21T13:08:29Z" level=info msg="Alloc=6248 TotalAlloc=67136 Sys=72128 NumGC=68 Goroutines=181"
time="2020-10-21T13:08:39Z" level=info msg="Performing periodic workflow GC"
time="2020-10-21T13:08:39Z" level=info msg="Zero old offloads, nothing to do"
time="2020-10-21T13:13:29Z" level=info msg="Alloc=6140 TotalAlloc=67444 Sys=72128 NumGC=71 Goroutines=181"
time="2020-10-21T13:13:39Z" level=info msg="Performing periodic workflow GC"
time="2020-10-21T13:13:39Z" level=info msg="Zero old offloads, nothing to do"
```
---

**Message from the maintainers**:

Impacted by this bug? Give it a 👍. We prioritise the issues with the most 👍.

### PR DESCRIPTION
**Pull request by alexec**
Signed-off-by: Alex Collins <alex_collins@intuit.com>

Checklist:

* [x] Either (a) I've created an [enhancement proposal](https://github.com/argoproj/argo/issues/new/choose) and discussed it with the community, (b) this is a bug fix, or (c) this is a chore.
* [x] The title of the PR is (a) [conventional](https://www.conventionalcommits.org/en/v1.0.0/), (b) states what changed, and (c) suffixes the related issues number. E.g. `"fix(controller): Updates such and such. Fixes #1234"`.  
* [x] My organization is added to [USERS.md](https://github.com/argoproj/argo/blob/master/USERS.md).
* [x] I've signed the CLA.
* [ ] ~~I have written unit and/or e2e tests for my change. PRs without these are unlikely to be merged.~~
* [x] My builds are green. Try syncing with master if they are not. 

Fixes #4338

### CODE REVIEW

File: server/artifacts/artifact_server.go [hunk 1]
```diff
@@ -137,8 +137,7 @@ func (a *ArtifactServer) getArtifact(ctx context.Context, wf *wfv1.Workflow, nod
 	if err != nil {
 		return nil, err
 	}
-
-	tmp, err := ioutil.TempFile(".", "artifact")
+	tmp, err := ioutil.TempFile("/tmp", "artifact")
```
Comment (sarabala1979) [inline server/artifacts/artifact_server.go:140]: Will it be automatically deleted? Or do we need to delete it? 
https://golang.org/pkg/io/ioutil/#TempFile

Comment (alexec) [reply server/artifacts/artifact_server.go:140]: yes - we should fix that bug too

**Review by sarabala1979 — APPROVED** [review]

**Comment by alexec** [comment]
Back-ported to v2.11

File: server/artifacts/artifact_server.go (hunk 1 above)
Comment (ekesken) [reply server/artifacts/artifact_server.go:140]: this change broke the output artifact downloads for us, we upgraded from 2.11.5 to 2.11.7, and we began to get errors like this:

`open /tmp/artifact058606664: no such file or directory`

reverting the upgrade fixed the downloads.

Comment (ckannon) [reply server/artifacts/artifact_server.go:140]: @ekesken - likely using runAsNonRoot? You'll need to mount /tmp as an emptyDir volume on argo-server pod so you can write to it. This path should probably be configurable as an argument.

Comment (ekesken) [reply server/artifacts/artifact_server.go:140]: I see that the fix in helm chart is already merged here: https://github.com/argoproj/argo-helm/commit/9939da51cfabe465ccb730a88976ce2a6b26112e

It was surprising to see a non-backward compatible change in a patch version change, that's why I wanted to point it here.

**Pull request merged**